test/suite/test_salvage01.py

```diff
@@ -317,7 +317,9 @@ def test_salvage_api_damaged(self):
         self.damage(self.tablename)
 
         # damage() closed the session/connection, reopen them now.
-        self.open_conn()
+        # Disable pre-fetching on the connection and its sessions as it is
+        # not expected to work correctly with damaged contents.
+        self.open_conn(config='prefetch=(available=false,default=false)')
```

Comment:
I might be misunderstanding something here. In the real world, this would correspond to a user trying to salvage their data, but needing to turn off some unrelated feature so it works properly.

If that take is correct - it's prefetch's job to not break existing salvage use cases.

Reply:
@korteland That totally makes sense. Yes, if the user goes back and does a salvage operation then pre-fetch is capable of catching this and turning pre-fetching off automatically but unfortunately that's not the case here. Would have to go back and think about how to make this work as it's not straightforward.